xcube_clms/store.py

```diff
@@ -0,0 +1,87 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 by the xcube development team and contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import logging
+from abc import ABC
+from typing import Tuple
+
+import xarray as xr
+from xcube.core.store import (
+    DataDescriptor,
+    DataStore,
+    DataTypeLike
+)
+from xcube.util.jsonschema import (
+    JsonObjectSchema,
+)
+
+from .clms import CLMS
+
+_LOG = logging.getLogger("xcube")
+
+
+class CLMSDataStore(DataStore, ABC):
+    """ CLMS implementation of the data store.
+    Args:
+        DataStore (xcube.core.store.DataStore): data store defined
+            in the xcube data store framework.
+    Returns:
+        CLMSDataStore: data store defined in the ``xcube_clms`` plugin.
+    """
+
+    def __init__(self, **clms_kwargs):
+        self.clms = CLMS(**clms_kwargs)
+
+
+    @classmethod
+    def get_data_store_params_schema(cls) -> JsonObjectSchema:
+        raise NotImplementedError
```

Comment:
Apparently this needs to be implemented so that we can use `new_data_store` function from xcube. Change to: 
```
    @classmethod
    def get_data_store_params_schema(cls) -> JsonObjectSchema:
        params = dict(
            url=JsonStringSchema(
                title="CMEMS Username",
                description="Preferably set by environment variable CMEMS_USERNAME ",
            )
        )
        return JsonObjectSchema(
            properties=dict(**params), required=None, additional_properties=False
        )
``